pages/app/add-testimonial.tsx

```diff
@@ -0,0 +1,112 @@
+import {
+  Input,
+  Textarea,
+  Button,
+  Heading,
+  FormControl,
+  FormLabel,
+  useToast,
+} from '@chakra-ui/react';
+import { ChangeEvent, FormEvent, useState } from 'react';
+
+import Section from '@packages/components/Section';
+import { useI18n } from '@packages/features/i18n-context';
+import { addTestimonial } from '@packages/services/testimonials';
+
+export default function AddTestimonialPage() {
+  const { t } = useI18n('testimonials');
+  const [isLoading, setIsLoading] = useState(false);
+  const [formState, setFormState] = useState({
+    name: '',
+    gitUsername: '',
+    testimonial: '',
+  });
+  const { name, gitUsername, testimonial } = formState;
+  const maxInputLength: number = 300;
+  const toast = useToast();
+
+  function onChange(
+    event: ChangeEvent<HTMLInputElement | HTMLTextAreaElement>,
+  ) {
+    setFormState({ ...formState, [event.target.name]: event.target.value });
+  }
+
+  function clearForm() {
+    setFormState({ name: '', gitUsername: '', testimonial: '' });
+  }
+
+  async function handleSubmit(e: FormEvent<HTMLFormElement>) {
+    e.preventDefault();
+    const isSubmitInvalid = name.length < 5 || testimonial.length < 20;
+
+    if (isSubmitInvalid)
+      toast({
+        description: t('toast.inputError'),
+        status: 'error',
+        isClosable: true,
+      });
+    else {
+      setIsLoading(true);
+      const error = await addTestimonial({
+        name,
+        testimonial,
+        gitUsername,
+      });
+      setIsLoading(false);
+
+      if (error) {
+        toast({
+          description: t('toast.invalidUserError'),
+          status: 'error',
+          isClosable: true,
+        });
+      } else {
+        toast({
+          description: t('toast.success'),
+          status: 'success',
+          isClosable: true,
+        });
+        clearForm();
+      }
+    }
+  }
```

Comment:
Fala meu mano, to curtindo muito os avanços, dei uma simplificada na function final, veja como simplifiquei a lógica dos if's e evitei os elses (código muito identado é um código de difícil leitura). 
também fiz uma simplificação dos toasts vou explicar no próximo comment

```suggestion
  async function handleSubmit(e: FormEvent<HTMLFormElement>) {
  async function handleSubmit(e: FormEvent<HTMLFormElement>) {
    e.preventDefault();
    const isSubmitInvalid = name.length < 5 || testimonial.length < 20;

    if (isSubmitInvalid) return errorToast();

    setIsLoading(true);
    const error = await addTestimonial({
      name,
      testimonial,
      gitUsername,
    });
    setIsLoading(false);

    if (error) return invalidUserToast();

    successToast();
    clearForm();
  }
```

Reply:
A fim de simplificar os toasts criei um novo customHook em `packages/hooks/useCustomToast.tsx`:

```jsx
import { useToast } from '@chakra-ui/react';

import { useI18n } from '@packages/features/i18n-context';

export default function useCustomToast() {
  const toast = useToast();
  const { t } = useI18n('testimonials');

  const errorToast = () => {
    toast({
      description: t('toast.inputError'),
      status: 'error',
      isClosable: true,
    });
  };

  const invalidUserToast = () => {
    toast({
      description: t('toast.invalidUserError'),
      status: 'error',
      isClosable: true,
    });
  };

  const successToast = () => {
    toast({
      description: t('toast.success'),
      status: 'success',
      isClosable: true,
    });
  };

  return {
    errorToast,
    invalidUserToast,
    successToast,
  };
}
```

Com este novo hook fiz o import assim no file original:

```jsx
import useCustomToast from '@packages/utils/useCustomToast';
const { errorToast, successToast, invalidUserToast } = useCustomToast();
```

Assim evitamos muita escrita no componente, e a lógica de submit fica bem clara, oque acha?